nodes/3d_armor_stand.lua

```diff
@@ -6,17 +6,58 @@ local S = wrench.translator
 local def = minetest.registered_nodes["3d_armor_stand:armor_stand"]
 
 local add_entity_and_node = def and def.after_place_node
-local update_entity = def and def.on_metadata_inventory_put
+local update_entity = def and def.on_metadata_inventory_take
 
-local lists = {
+-- detect which version is being used
+-- commit 8f775a64f0a29afbcc63cbcd78938c17609ed187 merged all lists into one
+local has_single_inventory = false
+for _, lbm in ipairs(core.registered_lbms) do
+	if "3d_armor_stand:update_inventories" == lbm.name then
+		has_single_inventory = true
+		break
+	end
+end
+
+local legacy_lists = {
 	"armor_head",
 	"armor_torso",
 	"armor_legs",
 	"armor_feet",
 }
 
+local lists = has_single_inventory and { "main" } or legacy_lists
+
 local function after_place(pos, player)
 	add_entity_and_node(pos, player)
+	local meta = core.get_meta(pos)
+	local inv = meta:get_inventory()
+	if has_single_inventory then
+		-- Server is running newer version with single inv-list.
+		-- Items may need to be moved from separate lists and old lists removed.
+		local temp_list
+		for i, list_name in ipairs(legacy_lists) do
+			temp_list = inv:get_list(list_name)
+			-- If first legacy list doesn't exist,
+			-- assume this stand was already updated before it was picked up.
+			if not temp_list then break end
+
+			-- Move to main list.
+			inv:set_stack("main", i, temp_list[1])
+			-- Remove legacy list.
+			inv:set_size(list_name, 0)
```

Comment:
Not everything needs a comment :)
```suggestion
			if not temp_list then
				break
			end
			inv:set_stack("main", i, temp_list[1])
			inv:set_size(list_name, 0)
```